### ISSUE TITLE
Add CMake build system in tf-psa-crypto directory

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/mbedtls#9283]
### Suggested enhancement
In preparation of the split, add a CMake build system to build the crypto library and the crypto test suites in the `tf-psa-crypto` directory. It should be an adaptation of the Mbed TLS CMake build system to the crypto code in `tf-psa-crypto` directory (see also the CMake build system in TF-PSA-Crypto): one CMakeLists.txt file in `tf-psa-crypto`, `tf-psa-crypto/core`, `tf-psa-crypto/tests`. Adapt the mbedtls CMake build system to use the TF-PSA-Crypto CMake build system to build its crypto library.

Depends on #9261, #9262, #9263, and #9266. 

DoD:
. be able to build the crypto library, the crypto test suites with CMake and be able to run ctest.
. documentation, packaging, installation not in scope of this issue.
. no reference to mbedtls tree not in scope of this issue

### Justification
Mbed TLS needs this for the repo split.

**Comment by irwir** [comment]
> `tf-psa-crypto/programs/psa`, `tf-psa-crypto/tests`

Since there is no library code in these subdirectories, should it be moved to root level `tests` or elsewhere?
Previous structure had `3rdparty`, `include` and `library` with pure library code and this was more logical and convenient for handling.
